spec/classes/puppet_server_puppetserver_spec.rb

```diff
@@ -586,6 +586,53 @@
         let(:params) { super().merge(server_puppetserver_version: '2.1.0') }
         it { should raise_error(Puppet::Error, /puppetserver <2.2 is not supported by this module version/) }
       end
+
+      describe 'allow jetty specific server threads' do
+        context 'with thread config' do
+          let :params do
+            {
+              server:                      true,
+              server_implementation:       'puppetserver',
+              server_ca:                   true,
+              server_puppetserver_dir:     '/etc/custom/puppetserver',
+              server_jruby_gem_home:       '/opt/puppetlabs/server/data/puppetserver/jruby-gems',
+              server_selector_threads:     1,
+              server_acceptor_threads:     2,
+              server_ssl_selector_threads: 3,
+              server_ssl_acceptor_threads: 4,
+              server_max_threads:          5
+            }
+          end
+
+          it { is_expected.to compile.with_all_deps }
+          it { is_expected.to contain_file('/etc/custom/puppetserver/conf.d/webserver.conf').
+               with_content(/selector-threads: 1/).
+               with_content(/acceptor-threads: 2/).
+               with_content(/ssl-selector-threads: 3/).
+               with_content(/ssl-acceptor-threads: 4/).
+               with_content(/max-threads: 5/)
+          }
+        end
+        context 'without thread config' do
+          let :params do
+            {
+              server:                  true,
+              server_implementation:   'puppetserver',
+              server_ca:               true,
+              server_puppetserver_dir: '/etc/custom/puppetserver',
+              server_jruby_gem_home:   '/opt/puppetlabs/server/data/puppetserver/jruby-gems'
+            }
+          end
+
+          it { is_expected.to compile.with_all_deps }
```

Comment:
Could you move this to the `with default parameters` block (starts at line 29)? Perhaps merge it in the existing check for this file at line 73.

Reply:
done